app/src/main/java/com/android/solvit/shared/ui/authentication/SignUp.kt

```diff
@@ -94,16 +112,44 @@ fun SignUpScreen(
           authViewModel, { navigationActions.navigateTo(Screen.SIGN_UP_CHOOSE_ROLE) }, {})
   val token = stringResource(R.string.default_web_client_id)
 
-  val goodFormEmail =
-      email.isNotBlank() &&
-          Patterns.EMAIL_ADDRESS.matcher(email).matches() &&
-          email.contains(".") &&
-          email.contains("@")
+  val emailRegex = Regex("^[\\w.-]+@[\\w.-]+\\.[a-zA-Z]{2,}$")
+  val goodFormEmail = emailRegex.matches(email)
+
   val passwordLengthComplete = password.length >= 6
   val samePassword = password == confirmPassword
 
   val isFormComplete = goodFormEmail && passwordLengthComplete && samePassword
 
+  val passwordApiService = createPasswordService()
+  val scope = rememberCoroutineScope()
+  val clipboardManager = LocalClipboardManager.current
+
+  val generatePassword: () -> Unit = {
```

Comment:
Would be good to see a comment explaining the security requirements for the generated passwords (length, complexity, etc.)